shared/macros/01-general.jinja

```diff
@@ -246,12 +253,18 @@ Therefore, you need to use a tool that can query the OCP API, retrieve the follo
 :type service: str
 
 #}}
-{{%- macro describe_firewalld_allow(port, proto, service) %}}
-    {{% if port %}}
+{{%- macro describe_firewalld_allow(port=none, proto=none, service=none) %}}
+    {{%- if ((port is none) and (proto is not none)) or ((port is not none) and (proto is none)) %}}
+    {{{- raise("describe_firewalld_allow: if you have proto/port you need respectively port/proto") }}}
+    {{%- endif %}}
+    {{%- if (service is none) and (port is none) %}}
+    {{{- raise("describe_firewalld_allow: define either service, or port+proto") }}}
+    {{%- endif %}}
+    {{% if port is not none %}}
```

Comment:
This forest of ifs is quite complex, and we have just a couple of proto/service and one proto/port. Would it work for you to split the macro instead? Not only it has to figure out how it was called at the beginning, but the effective body is also a one big if/else.

Reply:
Ok, it simplifies.